common/quotas/global/algorithm/scenarios_test.go

```diff
@@ -0,0 +1,478 @@
+
+					// and now: allow each host to use part of the unused rps (each host decides blindly because they're distributed)
+					//
+					// this can be done entirely on the limiter side, comparing either:
+					// - (targetRPS - usedRPS) / numHostsInRing    (does not require change in exchanged data currently, stays near limit)
+					// - (targetRPS - usedRPS) / numHostsInvolved  (requires additional data to be returned, can approach 2x limit)
+					//
+					// it could also be done on agg side, but the limiter already needs to know rps and this does not *quite* need it in the agg.
+					splitUnused := max(0, (targetRPS-allRps)/3)
+					local := targetRPS / 3.0
+					boostUnused := func(fair float64, used float64) float64 {
+						if fair < local {
+							// if allowed less than fallback local value, add unused up to local fallback.
+							// this helps low-weight hosts accept requests when there is unused RPS.
+							return min(local, fair+splitUnused)
+						} else {
+							// weighted higher than the average, trust the rps we get
+							return fair
+						}
+					}
+
+					// the only change on top of the original
+					results.targetRPS[0] = boostUnused(fair0, float64(r[0]))
+					results.targetRPS[1] = boostUnused(fair1, float64(r[1]))
+					results.targetRPS[2] = boostUnused(fair2, float64(r[2]))
+					t.Logf("round %d: "+
+						"calls: [%d, %d, %d], "+
+						"real: [%0.3f, %0.3f, %0.3f], "+
+						"unused: %0.3f (split %0.3f), "+
+						"original rps: [%0.3f, %0.3f, %0.3f], "+
+						"adjusted rps: [%0.3f, %0.3f, %0.3f] ",
+						ri,
+						r[0], r[1], r[2],
+						rps1, rps2, rps3,
+						targetRPS-allRps, splitUnused,
+						fair0, fair1, fair2, // same as original logic
+						results.targetRPS[0], results.targetRPS[1], results.targetRPS[2])
+				}
+				results.accepts = divide(results.accepts, len(round))
+				results.rejects = divide(results.rejects, len(round))
+				results.callRPS = divide(results.callRPS, len(round))
+				results.cumulativeRPS = results.targetRPS[0] + results.targetRPS[1] + results.targetRPS[2]
+			}
+			return
+		}
+
+		t.Run("zeros", func(t *testing.T) {
+			check(t, data{
+				targetRPS:     [3]float64{3.38, 5.00, 10.7},
+				callRPS:       [3]float64{1.25, 2.50, 5.00},
+				accepts:       [3]float64{0.75, 2.50, 5.00}, // very close, a main motivating reason for this approach
+				rejects:       [3]float64{0.50, 0.00, 0.00},
+				cumulativeRPS: targetRPS + 4.1, // when very low, allows moderate bit above
+			}, run(t, roundSchedule(0, 5)))
+		})
+		t.Run("low nonzeros", func(t *testing.T) {
+
+			check(t, data{
+				targetRPS:     [3]float64{4.34, 5.00, 8.72},
+				callRPS:       [3]float64{2.00, 3.00, 5.00},
+				accepts:       [3]float64{1.75, 3.00, 5.00}, // almost perfect, a main motivating reason for this approach
+				rejects:       [3]float64{0.25, 0.00, 0.00},
+				cumulativeRPS: targetRPS + 3.1, // less overage but still moderate
+			}, run(t, roundSchedule(1, 5)))
+		})
```

Comment:
and this is my proposed solution

Reply:
Could you elaborate on how a test scenario is a solution?

Reply:
fewer rejects, more accepts